core/iwasm/common/wasm_c_api.c

```diff
@@ -2986,6 +2998,12 @@ wasm_module_get_name(wasm_module_t *module)
     return wasm_runtime_get_module_name(module_ex->module_comm_rt);
 }
 
+bool
+wasm_module_is_underlying_binary_freeable(const wasm_module_t *module)
+{
+    return ((wasm_module_ex_t *)module)->is_binary_cloned;
```

Comment:
i'd suggest to make this check wasm_runtime_is_underlying_binary_freeable on the module as well.